File: SourceCppProject/sources/Services/AppLaunchConfiguration.cpp
#include "AppLaunchConfiguration.hpp"

#include <cstring>
#include <string>


AppLaunchConfiguration::AppLaunchConfiguration(int argc, char* argv[])
{
    for (int arg_index = 1; arg_index < argc; arg_index++)
    {
        std::string current_argument {argv[arg_index]};
        m_arguments_of_sources.push_back(current_argument);
    }
}

const std::vector<std::string> AppLaunchConfiguration::getArgumentsOfSources() const
{
    return m_arguments_of_sources;
}

File: TextLinesComparing.Library/AppLaunchConfiguration.cs
namespace TextLinesComparing.Library;

public static class AppLaunchConfiguration
{
    public static List<string> GetArgumentsOfSources(string[] arguments)
    {
        List<string> argumentsOfSources = new();

        for (int arg_index = 1; arg_index < arguments.Length; arg_index++)
        {
            argumentsOfSources.Add(arguments[arg_index]);
        }

        return argumentsOfSources;
    }
}
